test/models-resolve.test.js

```diff
@@ -0,0 +1,141 @@
+// Copyright IBM Corp. 2016. All Rights Reserved.
+// Node module: loopback-connector-remote
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+'use strict';
+
+const assert = require('assert');
+const helper = require('./helper');
+const loopback = require('loopback');
+const sinon = require('sinon');
+
+const RemoteConnector = require('../lib/remote-connector');
+
+describe('Model Resolve Tests', function() {
+  let serverApp, clientApp, remoteDs, resolveSpy, ChildModel;
+
+  before('spy remote connector resolve() function', () => {
+    resolveSpy = sinon.spy(RemoteConnector.prototype, 'resolve');
+  });
+
+  beforeEach('reset the resolve()', () => {
+    resolveSpy.reset();
+  });
+
+  after('restore remote connector resolve() function', () => {
+    resolveSpy.restore();
+  });
+
+  before('create remote datasource', () => {
+    serverApp = helper.createRestAppAndListen();
+    clientApp = loopback({localRegistry: true});
+    remoteDs = helper.createRemoteDataSource(clientApp, serverApp);
+  });
+
+  before('create a child model', () => {
+    const db = helper.createMemoryDataSource(clientApp);
+    ChildModel = clientApp.registry.createModel({
+      name: 'ChildModel',
+    });
+    clientApp.model(ChildModel, {dataSource: db});
```

Comment:
I find it rather unusual to have a relation between a ChildModel that's attached to (persisted in) a local in-memory datasource, and a RemoteModel that's attached to a remote datasource on the server. Is that intentional? Did you perhaps mean to attach ChildModel to remoteDs too?